src/Razor/test/RazorLanguage.Test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -1828,12 +1846,12 @@ public class SomeComponent : ComponentBase
 @using static Test2.SomeComponent
 @using Foo = Test3
 <MyComponent />
-<SomeComponent /> <!-- Not a component -->");
+<SomeComponent /> <!-- Not a component -->", throwOnFailure: false);
 
             // Assert
             AssertDocumentNodeMatchesBaseline(generated.CodeDocument);
             AssertCSharpDocumentMatchesBaseline(generated.CodeDocument);
-            var result = CompileToAssembly(generated);
+            CompileToAssembly(generated, throwOnFailure: false);
```

Comment:
Validate the diagnostic?

Reply:
It is already validated by testname.diagnostics.txt